topdown/save.go

```diff
@@ -317,6 +332,25 @@ func (s *saveSupport) Insert(path ast.Ref, rule *ast.Rule) {
 	module.Rules = append(module.Rules, rule)
 }
 
+func splitPackageAndRule(path ast.Ref) (ast.Ref, ast.Ref) {
+	p := path.Copy()
+
+	ruleRefStart := 2 // path always contains at least 3 terms (data. + one term in package + rule name)
+	for i := ruleRefStart; i < len(p.StringPrefix()); i++ {
+		t := p[i]
+		if str, ok := t.Value.(ast.String); ok && ast.IsVarCompatibleString(string(str)) {
```

Comment:
💭 We're turning an `ast.String` into a string to feed it into `ast.IsVarCompatibleString`. Perhaps the latter should just take an `ast.String`? They're both "ast things"...